fix: show --help message for CLI errors, add tests for delete

jsjoeio opened the pull request:
This PR shows a more helpful error message when calling `coder delete`.

## Subtasks

- [x] added tests

Fixes #1233

H/T to @mafredri for pointers & suggestions 🎉

_First time actually working in Go so please give any and all suggestions_

mafredri reviewed (commented):
Added some thoughts, in case they're helpful. I love that you're looking to fix this, it's not the best experience to have to dig into `--help` every time you've done something "wrong".

Perhaps an even better option than custom argument processing would be to automatically show the command help on error?

```
accepts 1 arg(s), received 0

Delete a workspace

Flags:
  -h, --help   help for delete

Use "coder delete [command] --help" for more information about a command.
```

The only problem I noticed here is that the `--help` is not helpful at all! The `Use:` description is not visible so it's impossible to know what the argument is 🤔.

mafredri reviewed (commented):
I noticed a limitation of our custom template for command help that may have contributed to part of the usability problem with the CLI. I think we should get that fixed first to better see what's missing.

jsjoeio committed:
feat(cli): add test for delete

This adds a new test for the `delete` command to ensure it works as
expected when provided the correct args.

jsjoeio committed:
fix(cli): use ExecuteC() to match Cobra

This modifies the `cli.Root().Execute()` to `cli.Root).ExecuteC()` to
match the default behavior of Cobra. We do this so errors will always
print the "run --help" line.

jsjoeio committed:
feat(cli): add WithoutParameters test for delete

This adds a new test to the `delete_test.go` suite to ensure the correct
behavior occurs when `delete` is called without an argument.

jsjoeio committed:
fixup! feat(cli): add WithoutParameters test for delete

jsjoeio committed:
refactor(cli): show --help error message on main

This adds an error message which shows when there is an error with any
commands called to improve the UX.

jsjoeio committed:
fixup! refactor(cli): show --help error message on main

jsjoeio committed:
refactor(cli): handle err with FormatCobraError

This adds a new helper function called `FormatCobraError` to `root.go`
so that we can colorize and add "--help" message to cobra command errors
like calling `delete`.

jsjoeio committed:
refactor(cli): add root_test.go, move delete test
